server.js

```diff
@@ -1,30 +1,75 @@
 import express from 'express'
 import bodyParser from 'body-parser'
 import cors from 'cors'
+import booksData from './data/books.json'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send('Possible endpoints /books | /books/pages?minPages=200 | /books/id/:id | /books/isbn/:isbn | /rating/:rating')
+})
+
+// Get all books
+app.get('/books', (req, res) => {
+  res.json(booksData)
+})
+
+// Filter books by minPages or maxPages
+// Query: ?minPages=NUM or ?maxPage=NUM
+app.get('/books/pages', (req, res) => {
+  const minPages = req.query.minPages 
+  const maxPages = req.query.maxPages 
+  const booksByMinPages = booksData.filter((item) => item.num_pages >= minPages)
+  const booksByMaxPages = booksData.filter((item) => item.num_pages <= maxPages)
+
+  if(minPages) {
+    res.json(booksByMinPages)
+  } 
+  if(maxPages) {
+    res.json(booksByMaxPages)
+  }
+  
+  res.json(booksData)
+})
```

Comment:
This is awesome and super useful for the user! It would be more RESTful to have this as a part of the `/books` endpoint but still keep both queries.

What happens if I want to search for a book with between 200 and 300 pages? An idea is to add a variable that you can refilter depending on the queries, and then in the end, just returning what's left:

```
let booksByPages = booksData

if(minPages) {
  booksByPages = booksByPages.filter((item) => item.num_pages >= minPages)
} 
if(maxPages) {
  booksByPages = booksByPages.filter((item) => item.num_pages <= maxPages)
}

res.json(booksByPages)

```